portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,17 +36,39 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private static final int REQUESTED_COMMENTS_LIMIT = 100;
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Sort comments by date from newest to oldest
+    int maxCommentsNumber;
+
+    // Check if max-comments parameter exists
+    if (request.getParameterMap().containsKey("max-comments")) {
+      // Try to convert the value from the 'max-comments' field of the query string to int
+      try {
+        maxCommentsNumber = Integer.parseInt(request.getParameter("max-comments"));
+      } catch (NumberFormatException e) {
+        maxCommentsNumber = REQUESTED_COMMENTS_LIMIT;
+      }
+    } else {
+      response.sendError(400, "'max-comments' parameter is missing.");
+      return;
+    }
+
+    // If maxCommentsNumber is negative, send error
+    if (maxCommentsNumber < 0) {
```

Comment:
What should they do when they get such an error message? It's hard for clients to link this error message with the problematic parameter. Also "max-comments" provides the maximum number of comments they want to load. The client doesn't intend to load negative number of comments. So we need to connect this error with the parameter so that clients know what's wrong. One optional idea: "max-comments cannot be negative".